Source/WebCore/page/Navigation.h

```diff
@@ -130,7 +130,7 @@ class Navigation final : public RefCounted<Navigation>, public EventTarget, publ
     ExceptionOr<void> updateCurrentEntry(UpdateCurrentEntryOptions&&);
 
     bool dispatchTraversalNavigateEvent(Ref<HistoryItem>);
-    bool dispatchPushReplaceReloadNavigateEvent(const URL&, NavigationNavigationType, bool isSameDocument);
+    bool dispatchPushReplaceReloadNavigateEvent(const URL&, NavigationNavigationType, URL oldURL);
```

Comment:
Why are we passing in a URL to calculate something instead of calculating it then passing in the bool result?
Also, why isn't it a const URL&?

Reply:
You are right, this is trying to do too much in one go. IIRC doing this may be more convenient for follow-up patches, but it is not needed now, so I'll revert this part of the PR.